src/main/scala/com/cognite/sdk/scala/v1/resources/files.scala

```diff
@@ -24,24 +30,58 @@ class Files[F[_]](val requestSession: RequestSession[F])
 
   implicit val errorOrFileDecoder: Decoder[Either[CdpApiError, File]] =
     EitherDecoder.eitherDecoder[CdpApiError, File]
+
+  override def createOne(item: FileCreate): F[File] =
+    requestSession
+      .send { request =>
+        request
+          .post(baseUri)
+          .body(item)
+          .response(asJson[Either[CdpApiError, File]])
+          .mapResponse {
+            case Left(value) => throw value.error
+            case Right(Left(cdpApiError)) =>
+              throw cdpApiError.asException(uri"$baseUri/byids")
+            case Right(Right(value)) => value
+          }
+      }
+
   override def createItems(items: Items[FileCreate]): F[Seq[File]] =
     items.items match {
-      case item :: Nil =>
-        requestSession
-          .send { request =>
-            request
-              .post(baseUri)
-              .body(item)
-              .response(asJson[Either[CdpApiError, File]])
-              .mapResponse {
-                case Left(value) => throw value.error
-                case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/byids")
-                case Right(Right(value)) => Seq(value)
-              }
-          }
+      case item :: Nil => {
+        createOne(item).map(Seq(_))
+      }
       case _ => throw new RuntimeException("Files only support creating one file per call")
     }
 
+  def upload(file: java.io.File): F[Either[String, File]] = {
+    implicit val sttpBackend: SttpBackend[Id, Nothing] = HttpURLConnectionBackend(
+      options = SttpBackendOptions.connectionTimeout(90.seconds)
+    )
+    val inputStream = new FileInputStream(file)
+    val item = FileCreate(name = file.getName())
+    val fileMetaData = createOne(item)
+    fileMetaData.map(
+      file =>
+        file.uploadUrl match {
+          case Some(uploadUrl) =>
+            sttp
+              .body(inputStream)
+              .put(uri"$uploadUrl")
+              .send()
+              .body
+              .map(_ => file)
+          case None =>
+            throw new SdkException("File upload did not return uploadUrl")
+        }
+    )
+  }
+
+  def uploadWithName(path: String, name: String): F[Either[String, File]] = {
```

Comment:
This method should take a `file: java.io.File` and a name, and allow you to pick the name to use in CDF freely (ie. doesn't have to match the file's name).

For example, `client.files.uploadWithName(new File("/path/to/some/data"), "my-favorite-foods.txt")`, and `client.files.upload(file: File) = client.files.uploadWithName(file, file.getName())` (or maybe `getCanonicalName` or something similar, please check what the Python SDK is doing with the filename and match that).